packages/amplify-schema-validator/src/validators/use-directives-from-newer-transformer-version.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  DocumentNode,
+  Kind,
+  ObjectTypeDefinitionNode,
+} from 'graphql';
+import { InvalidDirectiveError } from '../exceptions/invalid-directive-error';
+
+/**
+ * Validates that @belongsTo, @connection directives from older graphql transformer version is not used in version 1.0
+ *
+ * @param schema graphql schema
+ * @returns true if correct directives are used
+ */
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+const GRAPHQL_TRANSFORMER_V2_DIRECTIVES = ['hasOne', 'index', 'primaryKey', 'belongsTo', 'manyToMany', 'hasMany', 'default'];
+export const validateDirectivesFromNewerTransformerVersionAreNotUsed = (
```

Comment:
I suspect you could refactor the v1 directive validator and v2 directive validator into a helper function that accepts the necessary config, and returns the transformer, so we're not duplicating as much logic across the two validators.

Reply:
Refactored by taking out the common code